src/features/DebugSession.ts

```diff
@@ -80,6 +80,12 @@ export class DebugSessionFeature implements IFeature {
                     }
                 }
             }
+            else if (config.script) {
+                // In this case, the user has explicitly defined a script path
+                // so make sure to set the cwd to that path if the cwd wasn't
+                // explicitly set
+                config.cwd = config.cwd || config.script;
```

Comment:
That check is something we do on the PSES side.  I think around line 37, where we handle the no-folder workspace case we assign cwd there.  It would be the filepath if the document has been saved otherwise the workspace rootPath.  That leaves the unsaved file with a launch.json case.  That by default want's to use "${file}" for the cwd.  After line 53 we could set the CWD to the workspace root.  Then move the sanity check on CWD to the end of the `if (config.request === 'launch')`.  Does that make sense?